### PR TITLE
Fix issue #462 - Template id not returned as part of the error when an exception is thrown

### PR DESCRIPTION
**Pull request by ericmorand**
This commit fixes the issue by adding a 'file' property to Twig.Error and making sure that it is always set to the path of the file where the error is located.

Test suite has been updated accordingly.

### CODE REVIEW

File: test/templates/error/compile/entry.twig [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Should throw a compile error
+{{ "foo" bar }}
```
Comment (dave-irvine) [inline test/templates/error/compile/entry.twig]: Can you add the newlines at the end of these files?

File: test/test.rethrow.js [hunk 2]
```diff
@@ -1,23 +1,78 @@
...
+
+delete require.cache[path.resolve(path.join(__dirname, '../twig.js'))];
+
 var Twig = Twig || require("../twig"),
     twig = twig || Twig.twig;
 
-describe("Twig.js Rethrow ->", function() {
-    it("should throw a \"Unable to parse 'missing'\" exception", function() {
-        (function(){
+describe("Twig.js Rethrow ->", function () {
```
Comment (dave-irvine) [inline test/test.rethrow.js]: I think the rest of our tests use `function()` rather than `function ()`. Can you update accordingly?

Comment (ericmorand) [inline test/test.rethrow.js]: Oh yes, sorry about this. I fix this right now.

File: test/templates/error/compile/entry.twig (hunk 1 above)
Comment (ericmorand) [inline test/templates/error/compile/entry.twig]: Yep.

**Comment by ericmorand** [comment]
@dave-irvine That should be OK. Let me know if you want some more changes.

**Pull request merged**